fastplotlib/graphics/_base.py

```diff
@@ -68,3 +71,39 @@ def __repr__(self):
         else:
             return f"fastplotlib.{self.__class__.__name__} @ {hex(id(self))}"
 
+class Interaction(ABC):
+    @property
+    @abstractmethod
+    def indices(self) -> Any:
+        pass
+
+    @indices.setter
+    @abstractmethod
+    def indices(self, indices: Any):
+        pass
+
+    @property
+    @abstractmethod
+    def features(self) -> List[str]:
+        pass
+
+    @abstractmethod
+    def _set_feature(self, name: str, new_data: Any, indices: Any):
+        pass
+
+    @abstractmethod
+    def link(self, event: str, target: Graphic, feature: str, new_data: Any, indices_mapper: callable = None):
+        pass
+
+    def event_handler(self, event):
+        if event.type in self.registered_callbacks.keys():
+            for target_info in self.registered_callbacks[event.type]:
+                target_info.target._set_feature(feature=target_info.feature, new_data=target_info.new_data)
+
+@dataclass
+class EventData:
```

Comment:
this is really more of callback data than event data, because the event data comes from pygfx handling and `target`, `feature` and `new_data` are callback information.

Also this isn't how you construct a dataclass, you don't define `__init__` 
https://docs.python.org/3/library/dataclasses.html